mypy complains about griffe during precommit

Partly a followup from #744

I took another look at this whilst persuing another issue where I had needed a clean environment. This is on a fairly clean (no mellea dev) fedora 43 system.

* Changing pre-commit can allow all dependencies to be resolved, AND uv itself be installed, but the net result is creating an additional venv, it seems inappropriate to go to that extent

* I found that I only needed uv *installed* and in the *PATH* for pre-commit to work. I *did not* need to have a venv activated. No action needed, but adding as a relevant observation.

However this is where the griffe issue shows up

```➜  mellea-new git:(main) ✗ git status
On branch main
Your branch is up to date with 'origin/main'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   mellea/core/base.py

➜  mellea-new git:(main) ✗ pre-commit run
Ruff formatter...........................................................Passed
Ruff linter..............................................................Passed
MyPy.....................................................................Failed
- hook id: mypy
- exit code: 1

tooling/docs-autogen/decorate_api_mdx.py:410: error: Cannot find implementation or library stub for module named "griffe"  [import-not-found]
tooling/docs-autogen/audit_coverage.py:14: error: Cannot find implementation or library stub for module named "griffe"  [import-not-found]
tooling/docs-autogen/audit_coverage.py:14: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 2 errors in 2 files (checked 403 source files)

uv-lock..............................................(no files to check)Skipped
codespell................................................................Passed
➜  mellea-new git:(main) ✗ type uv
uv is /home/jonesn/.local/bin/uv
➜  mellea-new git:(main) ✗ type python
python is /usr/bin/python
➜  mellea-new git:(main) ✗
```

*  see also #456 - we can revisit type check tooling

cc: @jakelorocco @ajbozarth @avinash2692

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -19,7 +19,7 @@ repos:
     hooks:
       - id: mypy
         name: MyPy
-        entry: uv run --no-sync mypy --install-types --non-interactive . --exclude scratchpad/ --exclude tmp/
+        entry: uv run --no-sync mypy --install-types --non-interactive . 
         pass_filenames: false
         language: system
         types_or: [python, jupyter]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -288,6 +288,11 @@ install_types = true
 non_interactive = true
 disable_error_code = ["empty-body", "import-untyped"]
 python_version = "3.11"
+exclude = [
+"^tooling/",
+"^scratchpad/",
+"^tmp/"
+]
 
 [[tool.mypy.overrides]]
 # Keep import-not-found suppressed for optional dependencies

